internal/bminventory/inventory.go

```diff
@@ -1583,6 +1584,35 @@ func (b *bareMetalInventory) TransformClusterToDay2Internal(ctx context.Context,
 	return b.GetClusterInternal(ctx, installer.GetClusterParams{ClusterID: clusterID})
 }
 
+func (b *bareMetalInventory) GetClusterSupportedPlatformsInternal(ctx context.Context, params installer.GetClusterSupportedPlatformsParams) (*[]models.PlatformType, error) {
+	cluster, err := b.GetClusterInternal(ctx, installer.GetClusterParams{ClusterID: params.ClusterID})
+	if err != nil {
+		return nil, err
+	}
+
+	hostsPlatformsCounter := map[models.PlatformType]int{}
+	for _, h := range cluster.Hosts {
+		inventory, err := hostutil.UnmarshalInventory(h.Inventory)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, p := range *common.GetHostSupportedPlatforms(*inventory) {
+			hostsPlatformsCounter[p] += 1
```

Comment:
Why using a counter instead of intersection? please use funk. Intersect() and please implement all the logic in the platform.go